src/ifcparse/IfcSIPrefix.cpp

```diff
@@ -20,154 +20,194 @@
 #include "IfcSIPrefix.h"
 
 #ifdef HAS_SCHEMA_2x3
-#include "../ifcparse/Ifc2x3.h"
+#include "Ifc2x3.h"
 #endif
 #ifdef HAS_SCHEMA_4
-#include "../ifcparse/Ifc4.h"
+#include "Ifc4.h"
 #endif
 #ifdef HAS_SCHEMA_4x1
-#include "../ifcparse/Ifc4x1.h"
+#include "Ifc4x1.h"
 #endif
 #ifdef HAS_SCHEMA_4x2
-#include "../ifcparse/Ifc4x2.h"
+#include "Ifc4x2.h"
 #endif
 #ifdef HAS_SCHEMA_4x3_rc1
-#include "../ifcparse/Ifc4x3_rc1.h"
+#include "Ifc4x3_rc1.h"
 #endif
 #ifdef HAS_SCHEMA_4x3_rc2
-#include "../ifcparse/Ifc4x3_rc2.h"
+#include "Ifc4x3_rc2.h"
 #endif
 #ifdef HAS_SCHEMA_4x3_rc3
-#include "../ifcparse/Ifc4x3_rc3.h"
+#include "Ifc4x3_rc3.h"
 #endif
 #ifdef HAS_SCHEMA_4x3_rc4
-#include "../ifcparse/Ifc4x3_rc4.h"
+#include "Ifc4x3_rc4.h"
 #endif
 #ifdef HAS_SCHEMA_4x3
-#include "../ifcparse/Ifc4x3.h"
+#include "Ifc4x3.h"
 #endif
 #ifdef HAS_SCHEMA_4x3_tc1
-#include "../ifcparse/Ifc4x3_tc1.h"
+#include "Ifc4x3_tc1.h"
 #endif
 #ifdef HAS_SCHEMA_4x3_add1
-#include "../ifcparse/Ifc4x3_add1.h"
+#include "Ifc4x3_add1.h"
 #endif
 
 double IfcParse::IfcSIPrefixToValue(const std::string& v) {
-	if      ( v == "EXA"   ) return 1.e18;
-	else if ( v == "PETA"  ) return 1.e15;
-	else if ( v == "TERA"  ) return 1.e12;
-	else if ( v == "GIGA"  ) return 1.e9;
-	else if ( v == "MEGA"  ) return 1.e6;
-	else if ( v == "KILO"  ) return 1.e3;
-	else if ( v == "HECTO" ) return 1.e2;
-	else if ( v == "DECA"  ) return 1.e1;
-	else if ( v == "DECI"  ) return 1.e-1;
-	else if ( v == "CENTI" ) return 1.e-2;
-	else if ( v == "MILLI" ) return 1.e-3;
-	else if ( v == "MICRO" ) return 1.e-6;
-	else if ( v == "NANO"  ) return 1.e-9;
-	else if ( v == "PICO"  ) return 1.e-12;
-	else if ( v == "FEMTO" ) return 1.e-15;
-	else if ( v == "ATTO"  ) return 1.e-18;
-	else return 1.;
+    if (v == "EXA") return 1.e18;
+    else if (v == "PETA")
+        return 1.e15;
+    else if (v == "TERA")
+        return 1.e12;
+    else if (v == "GIGA")
+        return 1.e9;
+    else if (v == "MEGA")
+        return 1.e6;
+    else if (v == "KILO")
+        return 1.e3;
+    else if (v == "HECTO")
+        return 1.e2;
+    else if (v == "DECA")
+        return 1.e1;
+    else if (v == "DECI")
+        return 1.e-1;
+    else if (v == "CENTI")
+        return 1.e-2;
+    else if (v == "MILLI")
+        return 1.e-3;
+    else if (v == "MICRO")
+        return 1.e-6;
+    else if (v == "NANO")
+        return 1.e-9;
+    else if (v == "PICO")
+        return 1.e-12;
+    else if (v == "FEMTO")
+        return 1.e-15;
+    else if (v == "ATTO")
+        return 1.e-18;
+    else
+        return 1.;
 }
 
```

Comment:
to prevent a break after every `else if ()`this block needs to be commented with `// clang-format off`. Probably needs to tweak the setting `AllowShortIfStatementsOnASingleLine`.

Reply:
For me it's not an issue. This kind of formatting/alignment for readability tends to be a pain to maintain anyway. For me it's not necessary to preserve it.

I wonder though. Shouldn't we require braces around every if-statement/for-loop/etc. body? It's so easy to introduce errors (granted indentation would help you, but still):

```C++
if (something)
   do_something();
   do_something_else();
```

Reply:
There is an option to `InsertBraces`, but clang advises on extra caution to set this https://clang.llvm.org/docs/ClangFormatStyleOptions.html#insertbraces. I think the first format with this option could lead to falsely introduced braces, the following format runs should be fine.

Reply:
I would say let's give it a try. I don't know which kind of situations could give issues. I'd imagine they would result into compilation errors due to changes in scope, but yeah if you're unlucky maybe it results into a subtle behaviour changes. I can only think macro-based loops such as BOOST_FOREACH that would depend on more context than clang-format has, but that would then be a case of too little braces. I will review carefully.